moveit_ros/planning_interface/move_group_interface/src/move_group.cpp

```diff
@@ -132,19 +134,55 @@ class MoveGroup::MoveGroupImpl
 
     current_state_monitor_ = getSharedStateMonitor( robot_model_, tf_, node_handle_ );
 
+    ros::WallTime timeout_for_servers = ros::WallTime::now() + wait_for_servers;
+    if (wait_for_servers == ros::WallDuration())
+      timeout_for_servers == ros::WallTime(); // wait for ever
+    double allotted_time = wait_for_servers.toSec();
+
     move_action_client_.reset(new actionlib::SimpleActionClient<moveit_msgs::MoveGroupAction>
                               (node_handle_, move_group::MOVE_ACTION, false));
-    waitForAction(move_action_client_, wait_for_server, move_group::MOVE_ACTION);
+    waitForAction(move_action_client_, move_group::MOVE_ACTION, timeout_for_servers, allotted_time);
 
     pick_action_client_.reset(new actionlib::SimpleActionClient<moveit_msgs::PickupAction>
                               (node_handle_, move_group::PICKUP_ACTION, false));
-    waitForAction(pick_action_client_, wait_for_server, move_group::PICKUP_ACTION);
+    waitForAction(pick_action_client_, move_group::PICKUP_ACTION, timeout_for_servers, allotted_time);
 
     place_action_client_.reset(new actionlib::SimpleActionClient<moveit_msgs::PlaceAction>
                                (node_handle_, move_group::PLACE_ACTION, false));
-    waitForAction(place_action_client_, wait_for_server, move_group::PLACE_ACTION);
+    waitForAction(place_action_client_, move_group::PLACE_ACTION, timeout_for_servers, allotted_time);
+
+    execute_action_client_.reset(new actionlib::SimpleActionClient<moveit_msgs::ExecuteTrajectoryAction>
+                                 (node_handle_, move_group::EXECUTE_ACTION_NAME, false));
+    // TODO: standard waitForAction after deprecation period (L-turtle)
+    // waitForAction(execute_action_client_, move_group::EXECUTE_ACTION_NAME, timeout_for_servers, allotted_time);
 
+    // TODO: after deprecation period, i.e. for L-turtle, remove the following stuff
+    // SNIP
```

Comment:
move this SNIP-SNAP into helper function to hide all this extra compatibility logic?